### ISSUE TITLE
[FEATURE] Support of 3pid (medium: 'email') login

### ISSUE THREAD

**Issue by menturion** [syphon-org/syphon#512]
Currently, Syphon seems not to support the login via a 3pid (e.g. 'email'): 

```
{
  "type": "m.login.password",
  "identifier": {
    "type": "m.id.thirdparty",
    "medium": "email",
    "address": "<The third party address of the user>"
  },
  "password": "<password>",
}
```

This is essential for all homeservers that are using a password provider based on email + PWD.

**Comment by menturion** [comment]
### Addendum ###

This might be implemented in such a way that the current login form need not to be edited. By parsing the current user ID field input one would be able to identify and select the corresponding login type: 

```
if (isValidEmail(...)) {
   // Login using the "m.id.thirdparty" / "medium": "email"
} else if (isValidMsisdn(...)) {
   // Login using the "m.id.thirdparty" / "medium": "msisdn"
} else {
   // Current login method (user ID)
}
```

**Comment by EdGeraghty** [comment]
I have started laying the groundwork for this -> #518. @ereio feel free to comment on the direction I'm taking!

**Comment by EdGeraghty** [comment]
Implemented in #518 - will appear in 0.2.9!
